core/TypePtr.h

```diff
@@ -311,8 +311,10 @@ class TypePtr final {
         return toStringWithTabs(gs);
     }
 
+    struct ShowOptions {};
+
     // User visible type. Should exactly match what the user can write.
-    std::string show(const GlobalState &gs) const;
+    std::string show(const GlobalState &gs, const ShowOptions options = ShowOptions{}) const;
```

Comment:
Thoughts on passing `options` by reference?

Reply:
My plan was to keep it pretty small (a struct of bools), so copying should be fine. If it becomes a problem, we can revisit that decision.